### PR TITLE
Make killer & killable script builtins

### PR DESCRIPTION
**Pull request by wushin**
Only Death clears killer & killable
Move killer & killable to status namespace
closes #30

### CODE REVIEW

File: src/map/script-fun.cpp [hunk 1]
```diff
@@ -2175,6 +2175,30 @@ void builtin_pvpoff(ScriptState *st)
     }
 }
 
+static
+void builtin_killer(ScriptState *st)
+{
+    dumb_ptr<map_session_data> sd = script_rid2sd(st);
+    sd->state.killer = !sd->state.killer;
+
+    if (sd->state.killer)
```
Comment (mekolat) [inline src/map/script-fun.cpp]: @wushin can you please make this use arg0 so that we can do "killer 0;" and "killer 1;" ?
same for killable

:heart:

**Comment by mekolat** [comment]
*hugs wushin*

File: src/map/script-fun.cpp (hunk 1 above)
Comment (wushin) [inline src/map/script-fun.cpp]: I could but I have  couple of questions pertaining individual pvp settings.

**Comment by wushin** [comment]
@mekolat script builtins are 0/1 and @pvp is gm level zero, that toggles and disappears on death.

**Comment by wushin** [comment]
Now with Anti-Spam timer. So no one can rapidly toggle it on and off to grief people.

**Comment by mekolat** [comment]
@wushin I tested but players can turn it off mid-fight..

File: src/map/atcommand.cpp [hunk 2]
```diff
@@ -4030,15 +4030,33 @@ ATCE atcommand_character_storage_list(Session *s, dumb_ptr<map_session_data> sd,
...
+        ZString)
+{
+    if (sd->pvp_timer)
+        return ATCE::OKAY;
+
+    sd->state.pvpon = !sd->state.pvpon;
+    pc_setpvptimer(sd, battle_config.player_pvp_time);
+
+    if (sd->state.pvpon)
+        clif_displaymessage(s, "PvP: On"_s);
```
Comment (mekolat) [inline src/map/atcommand.cpp]: This should be sent to general tab. Please make it say "##3PvP : ##BOn" or something similar. The " : " will make it show in general tab, the ##3 will give it color and ##B will make it bold.

File: src/map/script-fun.cpp [hunk 3]
```diff
@@ -2175,6 +2175,42 @@ void builtin_pvpoff(ScriptState *st)
...
 }
 
+static
+void builtin_pvp(ScriptState *st)
+{
+    dumb_ptr<map_session_data> sd = script_rid2sd(st);
+    sd->state.pvpon = conv_num(st, &AARG(0));
+
+    if (sd->state.pvpon)
+        clif_displaymessage(sd->sess, "Pvp: On"_s);
```
Comment (mekolat) [inline src/map/script-fun.cpp]: why display messages on the pvp/killer/killable builtins? if the dev want to display a message then they'll manually add one

File: src/map/script-fun.cpp [hunk 4]
```diff
@@ -2175,6 +2175,42 @@ void builtin_pvpoff(ScriptState *st)
     }
 }
 
+static
+void builtin_pvp(ScriptState *st)
+{
+    dumb_ptr<map_session_data> sd = script_rid2sd(st);
+    sd->state.pvpon = conv_num(st, &AARG(0));
```
Comment (mekolat) [inline src/map/script-fun.cpp]: make it check arg0 because you could set it to any integer

**Comment by mekolat** [comment]
@wushin https://github.com/wushin/tmwa/pull/2

**Comment by wushin** [comment]
ok, changed all that.

**Comment by mekolat** [comment]
@wushin https://github.com/mekolat/tmwa/commit/13e2f1e30b1813ab4a028b37a34a4a5f4ecd5d23

**Comment by mekolat** [comment]
@wushin I rebased my PR: https://github.com/wushin/tmwa/pull/2

**Comment by wushin** [comment]
@mekolat too late I already rebased yours into mine.

**Comment by mekolat** [comment]
@wushin re-rebased t_t https://github.com/wushin/tmwa/pull/2

**Comment by mekolat** [comment]
@wushin why make the default channel reset on death? I don't see the need to make any channel reset on death

**Comment by mekolat** [comment]
@wushin can we remame getpvpflag to something else? it's no longer "flags"..

File: src/map/map.hpp [hunk 5]
```diff
@@ -142,11 +142,10 @@ struct map_session_data : block_list, SessionData
         unsigned shroud_disappears_on_pickup:1;
         unsigned shroud_disappears_on_talk:1;
         unsigned seen_motd:1;
+        unsigned pvpchannel;
```
Comment (mekolat) [inline src/map/map.hpp]: @wushin should there be :something at the end?

Comment (wushin) [inline src/map/map.hpp]: only if you want to limit the number of bits it packs.

**Pull request merged**